### ISSUE TITLE
[FR] [exec-on-event] re-execute exec after event command has finished

### ISSUE THREAD

**Issue by raffaem** [Alexays/Waybar#1960]
The docs for `exec-on-event` says:
> exec-on-event:
> typeof: bool
> default: true
> If an event command is set (e.g. on-click or on-scroll-up) then re-execute the script after executing the event command.

But it looks like the `exec` script is executed **immediately** after the event command is called.

I need it to be executed **after** the event command has exited instead.

This is because the event command performs some action and the `exec` script basically return the sucess/failure of those actions. But those actions need time to be undertaken.

Is it possible to run exec after the event command has finished?

If not, is it possible to implement this?

**Comment by raffaem** [comment]
Thanks for the workaround.

Should we leave this open as a bug on when "exec" script is called after an event?

**Comment by timblaktu** [comment]
Yes, thanks for the workaround. I hit this too, and have implemented the desired behavior sending the signal at the end of my `on-click` handler. 

I feel it should be left open, and only resolved if the documented behavior is implemented correctly or it is decided it won't be implemented and the docs changed.

**Comment by Karsteski** [comment]
Thanks for this. I've run across this issue as well, and @BoomerangNebula's suggestion is a useful workaround until the behaviour of events is changed

**Comment by jbirnick** [comment]
The same issue is mentioned here: https://github.com/Alexays/Waybar/issues/1681

Could you guys try to support this and raise some attention so that @Alexays _finally_ fixes this?

**Comment by mainrs** [comment]
> Could you guys try to support this and raise some attention so that @Alexays _finally_ fixes this?

@jbirnick  If you encounter a bug that you'd like to get fixed or you want a feature implemented, it's best to get involved yourself and submit a PR.

Putting pressure on open source maintainers is unhealthy. I'd like to refer to this part of [this gist](https://gist.github.com/richhickey/1563cddea1002958f96e7ba9519972d9):

> Open Source is Not About You
> 
> The only people entitled to say how open source 'ought' to work are people who run projects, and the scope of their entitlement extends only to their own projects.
> 
> Just because someone open sources something does not imply they owe the world a change in their status, focus and effort, e.g. from inventor to community manager.
> 
> As a user of something open source you are not thereby entitled to anything at all. You are not entitled to contribute. You are not entitled to features. You are not entitled to the attention of others. You are not entitled to having value attached to your complaints. You are not entitled to this explanation.
> 
> If you have expectations (of others) that aren't being met, those expectations are your own responsibility. You are responsible for your own needs. If you want things, make them.
> 
> Open source is a licensing and delivery mechanism, period. It means you get the source for software and the right to use and modify it. All social impositions associated with it, including the idea of 'community-driven-development' are part of a recently-invented mythology with little basis in how things actually work, a mythology that embodies, cult-like, both a lack of support for diversity in the ways things can work and a pervasive sense of communal entitlement.

You can always write your own patch and apply it before compiling the source code. If you dig a little bit you might find a good PR as your base.

**Comment by jbirnick** [comment]
@mainrs There is already a pull request for this (https://github.com/Alexays/Waybar/pull/1784, mentioned in the other issue which is again mentioned in my message above) which hasn't been merged _since over 2 years_, without any comment by the maintainer. So to me it seems like it actually is the case that the maintainer simply doesn't see this issue or doesn't realize its importance. That's why I'm just trying to reach him.

Your gist is right and a good reminder that I shouldn't overdo it, however, I think there _are_ certain expectations that you can have from an open source maintainer, such as the documentation being correct. This is not the case here. If the maintainer doesn't want the fix the issue, then at least they should correct the documentation to say what `exec-on-event` actually does.

**Comment by mainrs** [comment]
Having incorrect documentation is definitely something that is undesirable. I am also not a huge fan of that, but I can sympathize with people that just upload their projects to GitHub so others can take a look and use them. Sometimes the toll of that is just too much to bear. I went ahead and improved the wiki. The configuration field `exec-on-event` no specifically mentions that there are no guarantees that `exec` is executed after all `on-` event commands have finished. I also linked to #1784 as a possible (temporary) solution.

At least for this issue, #1784 is a possible patch that can be applied to the source code. I run Gentoo, and it build perfectly fine with the patch: https://github.com/Alexays/Waybar/pull/1784.patch

> Your gist is right and a good reminder that I shouldn't overdo it, and maybe I did so with the message that you quoted, so sorry for that.

My goal wasn't really to call you out or other people that I also shared the gist with. Sometimes, we, as users and developers, forget that it takes time, effort, motivation and many more to run an open source project. I just think that the gist is a good, albeit harsh, reminder of that.

I would like to see organizational changes to the repository by means of adding active members of the community as maintainers. This would allow to progressively merge PRs and remove the burden from Alexays.
